File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int ans = 0;
        int p1 = 0;
        int p2 = people.length - 1;
        int total = 0;
        
        while (p1 <= p2) {
            total = people[p1] + people[p2];
            
            if (total > limit) {
                ans++;
                p2--;
            } else if (total <= limit) {
                ans++;
                p2--;
                p1++;
            }
        }
        return ans;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => a - b);
    
    let ans = 0;
    let p1 = 0;
    let p2 = people.length - 1;
    
    while (p1 <= p2) {
        let sum = people[p1] + people[p2];
        
        if (sum <= limit) {
            ans++;
            p1++;
            p2--;
        } else {
            ans++;
            p2--;
        }
    }
    return ans;
}